### PR TITLE
Featureize move

### PR DESCRIPTION
**Pull request by sakridge**
#### Problem

Move takes a long time.

#### Summary of Changes

Move move into a feature.

This saves about 2 minutes for a clean build running setup.sh:

master: 4m4.437s
move-feature: 5m55.859s

Fixes #

### CODE REVIEW

**Comment by garious** [comment]
@sakridge, what's going on with this PR?

**Comment by garious** [comment]
@sakridge, ping

**Comment by sakridge** [comment]
@garious still working on it.

**Comment by TristanDebrunner** [comment]
Where exactly is this supposed to remove move from? `cargo build` and `cargo test` will still build all the move and libra crates because they are in the virtual manifest.

**Comment by sakridge** [comment]
> Where exactly is this supposed to remove move from? `cargo build` and `cargo test` will still build all the move and libra crates because they are in the virtual manifest.

So I think the biggest places for me is when you use the multinode-demo scripts with the default flags it should not be builidng/linking the move stuff in the normal flow of:
```
setup.sh
drone.sh
bootstrap-validator.sh
validator.sh
```

Similar place is when you use the `./net/net.sh restart.sh` or start or if you do `cargo test -p <package-name>` without --features=move.

Related to that is that it would be nice to have the feature plumbed through those scripts which this PR does not do. So if I pass --use-move to `./net/net.sh` then it enables the feature and creates a cluster with move enabled.

Also making sure all the move tests are exercised in CI with the move feature enabled. Those are in the move-loader-api, move-loader-program, and librapay-api. Also the bench-tps local_cluster tests with move mode.

@mvines any other place you can think of?

File: bench-tps/src/bench.rs [hunk 1]
```diff
@@ -314,21 +317,29 @@ fn generate_txs(
     let signing_start = Instant::now();
 
     let transactions = if let Some((
-        libra_genesis_keypair,
-        libra_pay_program_id,
+        _libra_genesis_keypair,
+        _libra_pay_program_id,
         _libra_mint_program_id,
-        libra_keys,
+        _libra_keys,
```
Comment (TristanDebrunner) [inline bench-tps/src/bench.rs]: Might be cleaner to `#[cfg_attr(not(feature = "move"), allow("unused_variables"))]`

File: core/Cargo.toml [hunk 2]
```diff
@@ -15,6 +15,7 @@ codecov = { repository = "solana-labs/solana", branch = "master", service = "git
 
 [features]
 pin_gpu_memory = []
+move = []
```
Comment (TristanDebrunner) [inline core/Cargo.toml]: This feature doesn't seem to actually be used.

Comment (sakridge) [reply core/Cargo.toml]:
```
error: failed to select a version for `solana-core`.
    ... required by package `solana-bench-tps v0.20.0 (/home/sakridge/src/solana/bench-tps)`
versions that meet the requirements `= 0.20.0` are: 0.20.0

the package `solana-bench-tps` depends on `solana-core`, with features: `move` but `solana-core` does not have these features.
```

Comment (TristanDebrunner) [reply core/Cargo.toml]: Remove `"solana-core/move"` at `bench-tps/Cargo.toml` Line 38

**Comment by garious** [comment]
Thanks!

**Pull request merged**